### PR TITLE
Initial port from private repo

### PR DESCRIPTION
**Pull request by iloveagent57**
This is the initial migration of prototype code from a private repo to the official repo.

### CODE REVIEW

File: openedx.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# This file describes this Open edX repo, as described in OEP-2:
+# http://open-edx-proposals.readthedocs.io/en/latest/oeps/oep-0002.html#specification
+
+nick: regis
+oeps:
+  oep-30:
+    state: True
+openedx-release: {ref: master}
+owner: edx/neem
+track-pulls: true
```
Comment (nedbat) [inline openedx.yaml]: We no longer need this key. Please remove it.

File: openedx.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# This file describes this Open edX repo, as described in OEP-2:
+# http://open-edx-proposals.readthedocs.io/en/latest/oeps/oep-0002.html#specification
+
+nick: regis
+oeps:
+  oep-30:
+    state: True
+openedx-release: {ref: master}
```
Comment (nedbat) [inline openedx.yaml]: This repo isn't yet meant to be in Open edX, so remove this key.

**Review by nedbat — DISMISSED** [review]

File: registrar/apps/enrollments/models.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    discovery_uuid = models.UUIDField(db_index=True, null=True)
+    title = models.CharField(max_length=255)
+    organizations = models.ManyToManyField(
+        Organization,
+        related_name='programs',
+        through='OrganizationProgramMembership'
+    )
+
+
+class OrganizationProgramMembership(TimeStampedModel):
```
Comment (kdmccormick) [inline registrar/apps/enrollments/models.py:34]: can a program belong to multiple organizations?

Comment (iloveagent57) [reply registrar/apps/enrollments/models.py:34]: Potentially, and that's why this is a many-to-many table b/w programs and orgs.

File: registrar/apps/enrollments/models.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    email = models.EmailField(db_index=True)
+    status = models.CharField(
+        max_length=64,
+        choices=STATUS_CHOICES,
+        default=PENDING,
+    )
+    history = HistoricalRecords()
+
+
+class LearnerProgramEnrollment(TimeStampedModel):
```
Comment (kdmccormick) [inline registrar/apps/enrollments/models.py:74]: Should `learner` and `program` be `unique_together` on this?

File: registrar/apps/enrollments/models.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+class CourseRun(TimeStampedModel):
+    """
+    Model that represents a course-discovery CourseRun entity.
+    """
+    discovery_uuid = models.UUIDField(db_index=True, null=True)
+    title = models.CharField(max_length=512)
+    key = models.CharField(max_length=255, db_index=True)
+
+
+class LearnerCourseRunEnrollment(TimeStampedModel):
```
Comment (kdmccormick) [inline registrar/apps/enrollments/models.py]: Should `learner` and `course_run` be `unique_together` on this?

File: registrar/apps/enrollments/models.py (hunk 4 above)
Comment (iloveagent57) [reply registrar/apps/enrollments/models.py:74]: Yes, good idea.

File: registrar/apps/enrollments/models.py (hunk 5 above)
Comment (iloveagent57) [reply registrar/apps/enrollments/models.py]: 👍

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
+version: "2.1"
+services:
+  registrar-db:
+    image: mysql:5.6
```
Comment (davestgermain) [inline docker-compose.yml:4]: I heard through the grapevine that postgresql is recommended for new IDAs that don't specifically need to use mysql. Have you thought about the choice here, or is this just inertia?

Comment (iloveagent57) [reply docker-compose.yml:4]: wha?!   I have not heard that rumor.

Comment (davestgermain) [reply docker-compose.yml:4]: I blame @nasthagiri for starting these rumors about postgresql. What do you say, @nasthagiri??

**Comment by kdmccormick** [comment]
first registrar :+1: ever

**Pull request merged**